### PR TITLE
Api

### PR DESCRIPTION
**Pull request by yurii-kovalchuk**
Set up axios wraper for the api endpoints.
Create several endpoints for external API: https://thecatapi.com/

### CODE REVIEW

File: src/api/api.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
+import axios from "axios";
+
+type Filter = {
+  limit?: number;
```
Comment (viktormovchaniuk) [inline src/api/api.tsx]: type QueryParams = Record<string, string>;

const queryParams: QueryParams = {
  param1: "value1",
  param2: "value2"
};

this type describe query params very well

**Review by viktormovchaniuk — COMMENTED** [review]
provide changes mentioned before pls. Then we can merge this and go further

File: src/api/api.tsx (hunk 1 above)
Comment (yurii-kovalchuk) [reply src/api/api.tsx]: it works, thanx

File: src/types/common.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+export type QueryParams = Record<string, string>;
+
+export type Breed = {
+  weight: {
```
Comment (viktormovchaniuk) [inline src/types/common.tsx]: maybe a good idea to create a specific types for nested objects values.
```
type Weight = {
  impertial: string,
  metric: string,
}
```

File: src/types/common.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  hairless: string;
+  natural: string;
+  rare: string;
+  rex: string;
+  suppressed_tail: string;
+  short_legs: string;
+  wikipedia_url: string;
+  hypoallergenic: string;
+  reference_image_id: string;
+  image: {
```
Comment (viktormovchaniuk) [inline src/types/common.tsx]: same as for Weight

**Review by viktormovchaniuk — COMMENTED** [review]
Great work! Everything looks good, but I do have a few suggestions to enhance the code further.

In my opinion, adhering to the DRY (Don't Repeat Yourself) principle is essential. It might be a bit redundant to create an additional type for each response type. Instead, consider creating a universal type like 'Image' and leverage TypeScript utilities to extend, replace, or modify it as needed. This approach can help you avoid redundancy down the line, especially if the entity structure on the back-end changes.

Think about your application in terms of entities (data) and their relationships. A good example of a type could be:

```
export type Vote = {
  id: string;
  image_id: string;
  sub_id: string | null;
  created_at: string;
  value: 1 | -1; // Consider using an Enum type or mb it is just number. Look at the GET response value
  country_code: string;
  image: Image;
  message: string;
};
```

It's usually better to avoid optional fields whenever possible. If you're dealing with different sets of fields for creating a response and getting a response, don't give a fck too much about it. You can either point your response as a common generic type or omit specifying the request response altogether.

Overall, I'm approving this MR as the work is commendable. You can decide how to implement these changes later, but they would be valuable for code maintainability and flexibility. Well done!

Also look at common commit suggestions: [conventionalcommits](https://www.conventionalcommits.org/en/v1.0.0/)
This links have a lot of common good habits in committing practice

And don't forget to squashe your commits when this MR will be merged!!!

File: src/types/common.tsx (hunk 2 above)
Comment (yurii-kovalchuk) [reply src/types/common.tsx]: done

File: src/types/common.tsx (hunk 3 above)
Comment (yurii-kovalchuk) [reply src/types/common.tsx]: also done

**Comment by yurii-kovalchuk** [comment]
Thanx for your feedback. I had considered your recommendations and made few changes with types - please check it up.

**Pull request merged**